docs/design/client-requests.md

```diff
@@ -4,61 +4,73 @@ sidebar_position: 3
+
+For the best performance, each request should batch as many events as possible. Typically this means
+funneling events through fewer client instances (e.g. a single client instance per process).
 
 The maximum number of events per batch depends on the maximum message size
-(`config.message_size_max`) and the operation type.
-(TODO: Expose batch size in the client instead).
+(`config.message_size_max`, specified in bytes) and the operation type. (TODO: Expose batch size in
+the client instead).
 
 In the default configuration, the batch sizes are:
 
-| Operation               | Batch Size |
-| ----------------------- | ---------: |
-| `lookup_accounts`       | 8190       |
-| `lookup_transfers`      | 8190       |
-| `create_accounts`       | 8190       |
-| `create_transfers`      | 8190       |
-| `get_account_transfers` | 8190       |
-| `get_account_history`   | 8190       |
+| Operation               | Batch Size (Bytes) |
+| ----------------------- | -----------------: |
+| `lookup_accounts`       |               8190 |
+| `lookup_transfers`      |               8190 |
+| `create_accounts`       |               8190 |
+| `create_transfers`      |               8190 |
+| `get_account_transfers` |               8190 |
+| `get_account_history`   |               8190 |
 
 The application layer can batch events by itself, however, when the same client instance is shared
-by multiple threads/tasks, it is able to transparently batch requests of the same operation type
-in a single message.
+by multiple threads/tasks, it is able to transparently batch requests of the same operation type in
+a single message.
+
+- [Node](/src/clients/node/README.md#batching)
+- [Go](/src/clients/go/README.md#batching)
+- [Java](/src/clients/java/README.md#batching)
+- [.NET](/src/clients/dotnet/README.md#batching)
 
-Read more about how two-phase transfers work with each client.
+## Example API Layer Architecture
 
-* [Node](/src/clients/node/README.md#batching)
-* [Go](/src/clients/go/README.md#batching)
-* [Java](/src/clients/java/README.md#batching)
-* [.NET](/src/clients/dotnet/README.md#batching)
+When building an application on TigerBeetle, you may want to use an intermediate API layer to batch
+events.
 
-## API Layer Architecture
+This is primarily applicable if the number of services that need to query TigerBeetle:
 
-In some application architectures, the number of services that need to query TigerBeetle may:
 - [exceed `config.clients_max`](./client-sessions.md#eviction), or
-- may require additional [batching](#batching-events) to optimize throughput.
+- require additional [batching](#batching-events) to optimize throughput.
+
+Rather than each service connecting to TigerBeetle directly, you can set up your application
+services to forward their requests to a pool of intermediate services, as illustrated below. This
+API layer can coalesce events from many application services into requests, and forward back the
+respective replies. 
```

Comment:
We should probably clarify that TigerBeetle does not provide this intermediate service or a client to pool/connect to them.